cmd/nanotube/listen.go

```diff
@@ -116,55 +124,30 @@ loop:
 	close(queue)
 }
 
-func listenUDP(conn *net.UDPConn, cfg *conf.Main, queue chan string, stop <-chan struct{}, lg *zap.Logger, ms *metrics.Prom) {
-
-	if cfg.UDPOSBufferSize != 0 {
-		err := conn.SetReadBuffer(int(cfg.UDPOSBufferSize))
-		if err != nil {
-			lg.Error("error setting UDP reader buffer size", zap.Error(err))
-		}
-	}
-
+func listenUDP(conn net.Conn, queue chan string, stop <-chan struct{},
+	UDPbBufSize int, lg *zap.Logger, ms *metrics.Prom) {
 	defer func() {
 		cerr := conn.Close()
 		if cerr != nil {
 			lg.Error("closing the incoming connection", zap.Error(cerr))
 		}
 	}()
 
-	// TODO (grzkv): Move out the length
-	in := make(chan *bytes.Buffer, 50)
-
-	go func() {
-		for {
-			buf := make([]byte, 64*1024)
-			nRead, cerr := conn.Read(buf)
-			buf = buf[:nRead]
-			if cerr != nil {
-				// TODO (grzkv) gather info
-				// https://github.com/golang/go/issues/4373
-				// ignore net: errClosing error as it will occur during shutdown
-				// if strings.HasSuffix(err.Error(), "use of closed network connection") {
-				// 	return
-				// }
-				ms.UDPReadFailures.Inc()
-			} else {
-				// TODO (grzkv): Reduce allocs
-				in <- bytes.NewBuffer(buf)
-			}
+	r := bufio.NewReaderSize(conn, UDPbBufSize)
+	for {
+		line, err := r.ReadBytes('\n')
+		if err != nil && err != io.EOF {
```

Comment:
Why this extra check for io.EOF error? -https://golang.org/pkg/bufio/#Reader.ReadBytes

Reply:
Because EOF is not an error in this case but normal behavior.